From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Wed, 5 Mar 2025 18:08:25 +0300
Subject: [PATCH] Configurable MC-59471

Bring back MC-59471, MC-129055 on 1.21.2+, which fixed in 1.21.2 snapshots 24w33a and 24w36a

P.S: This setting is different from skip-tripwire-hook-placement-validation in Paper

diff --git a/net/minecraft/world/level/block/TripWireHookBlock.java b/net/minecraft/world/level/block/TripWireHookBlock.java
index 8a3a8b0fdf9545a41501dc992c6982d9c8ce7b66..52fa81fdce9d7998f4c2d3b90dfc08391218f7fd 100644
--- a/net/minecraft/world/level/block/TripWireHookBlock.java
+++ b/net/minecraft/world/level/block/TripWireHookBlock.java
@@ -187,7 +187,6 @@ public class TripWireHookBlock extends Block {
             if (!cancelledEmitterHook) { // Paper - Call BlockRedstoneEvent
             emitState(level, pos, flag2, flag3, flag, flag1);
             if (!attaching) {
-                if (io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.skipTripwireHookPlacementValidation || level.getBlockState(pos).is(Blocks.TRIPWIRE_HOOK)) // Paper - Validate tripwire hook placement before update
                 level.setBlock(pos, blockState1.setValue(FACING, direction), 3);
                 if (shouldNotifyNeighbours) {
                     notifyNeighbors(block, level, pos, direction);
@@ -200,10 +199,18 @@ public class TripWireHookBlock extends Block {
                     BlockPos blockPos1 = pos.relative(direction, i2);
                     BlockState blockState2 = blockStates[i2];
                     if (blockState2 != null) {
-                        BlockState blockState3 = level.getBlockState(blockPos1);
-                        if (blockState3.is(Blocks.TRIPWIRE) || blockState3.is(Blocks.TRIPWIRE_HOOK)) {
-                            if (!io.papermc.paper.configuration.GlobalConfiguration.get().blockUpdates.disableTripwireUpdates || !blockState3.is(Blocks.TRIPWIRE))  level.setBlock(blockPos1, blockState2.trySetValue(ATTACHED, flag2), 3); // Paper - prevent tripwire from updating
+                        // DivineMC start - Configurable MC-59471
+                        if (level.divineConfig.allowTripwireDupe) {
+                            level.setBlock(blockPos1, blockState2.trySetValue(ATTACHED, flag2), 3);
+                            level.getBlockState(blockPos1);
+                        } else {
+                            BlockState blockState3 = level.getBlockState(blockPos1);
+                            if (blockState3.is(Blocks.TRIPWIRE) || blockState3.is(Blocks.TRIPWIRE_HOOK)) {
+                                if (!io.papermc.paper.configuration.GlobalConfiguration.get().blockUpdates.disableTripwireUpdates || !blockState3.is(Blocks.TRIPWIRE))
+                                    level.setBlock(blockPos1, blockState2.trySetValue(ATTACHED, flag2), 3); // Paper - prevent tripwire from updating
+                            }
                         }
+                        // DivineMC end - Configurable MC-59471
                     }
                 }
             }
